detectorCommon/src/xtalk_k4geo.cpp

```diff
@@ -0,0 +1,245 @@
+#include "detectorCommon/DetUtils_k4geo.h"
+#include "detectorCommon/xtalk_k4geo.h"
+
+// DD4hep
+#include "DDG4/Geant4Mapping.h"
+#include "DDG4/Geant4VolumeManager.h"
+
+// Geant
+#include "G4NavigationHistory.hh"
+
+// ROOT
+#include "TGeoBBox.h"
+
+#ifdef HAVE_GEANT4_UNITS
+#define MM_2_CM 1.0
+#else
+#define MM_2_CM 0.1
+#endif
+
+#include <iostream>
+
+#include <unordered_set>
+
+namespace det {
+namespace xtalk {
+
+// use it for module-theta merged readout (FCCSWGridModuleThetaMerged_k4geo)
+std::vector<std::pair<uint64_t, double>> xtalk_neighbours_ModuleThetaMerged(const dd4hep::DDSegmentation::FCCSWGridModuleThetaMerged_k4geo& aSeg,
+						   const dd4hep::DDSegmentation::BitFieldCoder& aDecoder,
+						   const std::vector<std::string>& aFieldNames,
+						   const std::vector<std::vector<std::pair<int, int>>>& aFieldExtremes_layer,
+						   uint64_t aCellId) {
+
+  std::vector<std::pair<uint64_t, double>> xtalk_neighbours;
+  
+  // check that field names and extremes have the proper length
+  if (aFieldNames.size() != 3 || aFieldExtremes_layer[0].size()!=3) {
+    std::cout << "ERROR: the vectors aFieldNames and aFieldSizes should be of length = 3, corresponding to the theta/module/layer fields" << std::endl;
+    std::cout << "ERROR: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }  
+  // find index of layer, module and theta in the field name vector
+  int idModuleField(-1);
+  int idThetaField(-1);
+  int idLayerField(-1);
+  for (uint itField = 0; itField < aFieldNames.size(); itField++) {
```

Comment:
Can you use just int and not bother with casts bellow?

Reply:
Corresponding modification has been made for l46-l54 and l225-l232